src/settings/language-and-timezone-settings.tsx

```diff
@@ -4,74 +4,95 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import React, { FC, useCallback, useMemo } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 
 import {
 	Container,
 	FormSubSection,
 	Modal,
 	Padding,
 	Select,
+	SelectItem,
 	SingleSelectionOnChange,
 	Text
 } from '@zextras/carbonio-design-system';
 import { find } from 'lodash';
 
-import { localeList, timeZoneList } from './components/utils';
+import {
+	localeList,
+	SettingsSectionProps,
+	timeZoneList,
+	upsertPrefOnUnsavedChanges
+} from './components/utils';
 import { timezoneAndLanguageSubSection } from './general-settings-sub-sections';
-import { AccountSettings, AddMod, PrefsMods } from '../../types';
+import { useReset } from './hooks/use-reset';
+import { AccountSettings, AddMod } from '../../types';
 import { getT } from '../store/i18n';
 
-const LanguageAndTimeZone: FC<{
+interface LanguageAndTimeZoneSettingsProps extends SettingsSectionProps {
 	settings: AccountSettings;
+	addMod: AddMod;
 	open: boolean;
 	setOpen: (arg: boolean) => void;
-	addMod: AddMod;
-}> = ({ settings, addMod, open, setOpen }) => {
+}
+
+export const LanguageAndTimeZoneSettings = ({
+	settings,
+	addMod,
+	open,
+	setOpen,
+	resetRef
+}: LanguageAndTimeZoneSettingsProps): React.JSX.Element => {
 	const t = getT();
 	const locales = useMemo(() => localeList(t), [t]);
 	const timezones = useMemo(() => timeZoneList(t), [t]);
 	const sectionTitle = useMemo(() => timezoneAndLanguageSubSection(t), [t]);
 
-	const updatePrefs = useCallback(
-		(prefValue: PrefsMods[keyof PrefsMods], prefKey: keyof PrefsMods) => {
-			addMod('prefs', prefKey, prefValue);
-		},
-		[addMod]
-	);
+	const updatePref = useMemo(() => upsertPrefOnUnsavedChanges(addMod), [addMod]);
 
-	const defaultLocale = useMemo(
-		() =>
-			(settings.prefs.zimbraPrefLocale &&
-				find(locales, { id: `${settings.prefs.zimbraPrefLocale}` })) ||
-			find(locales, { id: 'en' }),
-		[locales, settings.prefs.zimbraPrefLocale]
+	const [prefLocale, setPrefLocale] = useState<string>(settings.prefs.zimbraPrefLocale ?? '');
+
+	const prefLocaleSelectedValue = useMemo<SelectItem>(
+		() => find(locales, (item) => item.value === prefLocale) as SelectItem,
```

Comment:
Can we add a comment here with the ref of the issue with which we will remove the cast?